app/src/test/java/com/fstyle/structure_android/screen/main/MainPresenterTest.java

```diff
@@ -39,58 +37,53 @@
     Validator mValidator;
     @Mock
     CustomCompositeSubscription mSubscription;
+    @Mock
+    UserRepository mUserRepository;
 
-    private UserRepository mUserRepository;
     private MainPresenter mMainPresenter;
 
     @Before
     public void setUp() throws Exception {
         MockitoAnnotations.initMocks(this);
-        RxAndroidPlugins.getInstance().registerSchedulersHook(new RxAndroidSchedulersHook() {
-            @Override
-            public Scheduler getMainThreadScheduler() {
-                return Schedulers.immediate();
-            }
-        });
-        mUserRepository = new UserRepository(mLocalDataSource, mRemoteDataSource);
-        mMainPresenter = new MainPresenter(mUserRepository, mValidator, mSubscription);
+        mMainPresenter = new MainPresenter(mUserRepository, mValidator, mSubscription,
+                new ImmediateSchedulerProvider());
         mMainPresenter.setViewModel(mViewModel);
     }
 
-    @After
-    public void tearDown() {
-        RxAndroidPlugins.getInstance().reset();
-    }
-
     @Test
-    public void searchUsers() throws Exception {
+    public void searchUsers_ValidDataInput_ShowListUser() {
         // Give
         List<User> users = new ArrayList<>();
         users.add(new User(USER_LOGIN_1));
         users.add(new User(USER_LOGIN_2));
 
         // When
-        Mockito.when(mUserRepository.searchUsers(Mockito.anyInt(), Mockito.anyString()))
-                .thenReturn(Observable.just(users));
+        when(mUserRepository.searchUsers(Mockito.anyInt(), Mockito.anyString())).thenReturn(
+                Observable.just(users));
+        when(mValidator.validateAll(mViewModel, false)).thenReturn(true);
 
         // Then
         mMainPresenter.searchUsers(2, USER_LOGIN_1);
 
         Mockito.verify(mViewModel, Mockito.never()).onSearchError(null);
-        //        Mockito.verify(mView).showListUser(users);
+        Mockito.verify(mViewModel).onSearchUsersSuccess(users);
+    }
 
+    @Test
+    public void searchUsers_ValidDataInputNetworkError_ShowError() {
```

Comment:
ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions